back-end/src/routes/api.js

```diff
@@ -104,8 +97,203 @@ const uploadFile = (request, response) => {
+ */
+const getPhotoSocial = (request, response) => {
+  (async () => {
+    var statusCode = 400;
+    var message = "User does not exist";
+    console.debug(request.headers.email);
+    const reqEmail = request.headers.email;
+    // For now this will be in the headers. Modify later maybe.
+    if (reqEmail) {
+      try {
+        console.debug("Querying photos");
+        
+        // Query for information based on input email. Use email index to search.
+        var queryParams = {
+          TableName: "Files",
+          IndexName: "email-file-index",
+          KeyConditionExpression: "#email = :email",
+          ExpressionAttributeNames: {
+            "#email": "email"
+          },
+          ExpressionAttributeValues: {
+            ":email": reqEmail
+          }
+        };
+        const queryData = await dynamodb.query(queryParams).promise();
+        const items = queryData.Items;
+
+        var responseData = [];
+        // Loop through photos to send all object data
+        prevTitle = "";
+        for (let i = 0; i < items.length; i++) {
+          if (items[i].title === prevTitle) {
+            continue;
+          }
+
+          // Get picture data from S3 to send in response.
+          var getParams = {
+            Bucket: BUCKET_NAME,
+            Key: items[i].file
+          };
+          var photoData = await s3.getObject(getParams).promise();
+          
+          responseObject = {
+            photo: photoData.Body.buffer,
+            title: items[i].title,
+            desc: items[i].desc,
+            likes: items[i].likes,
+            uploadDate: items[i].uploadDate
+          };
+
+          responseData.push(responseObject);
+          prevTitle = items[i].title;
+        }
+        statusCode = 200;
+        console.debug(responseData);
+        response.status(statusCode).send(responseData);
+      }
+      catch(e) {
+        console.debug("Could not retrieve photos", e);
+      }
+      console.debug("Retrieval successful");
+    }
+    else {
+      response.status(statusCode).send(message);
+    }
+  })().catch(e => {
+    console.debug("Something went wrong");
+  });
+};
+
+/**
+ * Retrieve all photos from a specific tag.
+ * @param {tag: String} request 
+ * @param {[{photo: Object, uploadDate: String, title: String}]} response 
+ */
+const getPhotoByTag = (request, response) => {
+  (async () => {
+    var statusCode = 400;
+    var message = "Empty";
+    const reqTag = request.headers.tag;
+    if (reqTag) {
+      try {
+        var queryParams = {
+          TableName: "Files",
+          IndexName: "tag-file-index",
+          KeyConditionExpression: "#tag = :tag",
+          ExpressionAttributeNames: {
+            "#tag": "tag"
+          },
+          ExpressionAttributeValues: {
+            ":tag": reqTag
+          }
+        };
+
+        const queryData = await dynamodb.query(queryParams).promise();
+        const items = queryData.Items;
+        console.debug("Data Queried");
```

Comment:
do we need debug statements here

Reply:
Linter told me to use console.debug instead of console.log.

Reply:
Yes, however we shouldn't have a bunch of debug statements when our backend is running. see [here](https://dev.to/mornir/-dont-leave-console-logs-in-production-14na)